Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Instances.lean

```diff
@@ -32,19 +33,115 @@ continuous functional calculus, normal, selfadjoint
 
 noncomputable section
 
-section Normal
+local notation "σₙ" => quasispectrum
+local notation "σ" => spectrum
 
-instance IsStarNormal.cfc_map {R A : Type*} {p : A → Prop} [CommSemiring R] [StarRing R]
-    [MetricSpace R] [TopologicalSemiring R] [ContinuousStar R] [TopologicalSpace A] [Ring A]
-    [StarRing A] [Algebra R A] [ContinuousFunctionalCalculus R p] (a : A) (f : R → R) :
-    IsStarNormal (cfc f a) where
-  star_comm_self := by
-    rw [Commute, SemiconjBy]
-    by_cases h : ContinuousOn f (spectrum R a)
-    · rw [← cfc_star, ← cfc_mul .., ← cfc_mul ..]
-      congr! 2
-      exact mul_comm _ _
-    · simp [cfc_apply_of_not_continuousOn a h]
+/-!
+### Pull back a non-unital instance from a unital one on the unitization
+-/
+
+section RCLike
+
+variable {𝕜 A : Type*} [RCLike 𝕜] [NonUnitalNormedRing A] [StarRing A] [CstarRing A]
+variable [CompleteSpace A] [NormedSpace 𝕜 A] [IsScalarTower 𝕜 A A] [SMulCommClass 𝕜 A A]
+variable [StarModule 𝕜 A] {p : A → Prop} {p₁ : Unitization 𝕜 A → Prop}
+
+local postfix:max "⁺¹" => Unitization 𝕜
+
+variable (hp₁ : ∀ {x : A}, p₁ x ↔ p x) (a : A) (ha : p a)
+variable [ContinuousFunctionalCalculus 𝕜 p₁]
+
+open scoped ContinuousMapZero
+
+
+open Unitization in
+/--
+This is an auxiliary definition used for constructing an instance of the non-unital continuous
+functional calculus given a instance of the unital one on the unitization.
+
+This is the natural non-unital star homomorphism obtained from the chain
+```lean
+calc
+  C(σₙ 𝕜 a, 𝕜)₀ →⋆ₙₐ[𝕜] C(σₙ 𝕜 a, 𝕜) := ContinuousMapZero.toContinuousMapHom
+  _             ≃⋆[𝕜] C(σ 𝕜 (↑a : A⁺¹), 𝕜) := Homeomorph.compStarAlgEquiv'
+  _             →⋆ₐ[𝕜] A⁺¹ := cfcHom
+```
+This range of this map is contained in the range of `(↑) : A → A⁺¹` (see `nucfcAux_mem_range_inr`),
+and so we may restrict it to `A` to get the necessary homomorphism for the non-unital continuous
+functional calculus.
+-/
+noncomputable def nucfcAux : C(σₙ 𝕜 a, 𝕜)₀ →⋆ₙₐ[𝕜] A⁺¹ :=
+  (cfcHom (R := 𝕜) (hp₁.mpr ha) : C(σ 𝕜 (a : A⁺¹), 𝕜) →⋆ₙₐ[𝕜] A⁺¹) |>.comp
```

Comment:
Shouldn't we stick with `cfcₙ` throughout, rather than have a few random `nucfc`?